apps/src/codebridge/codebridgeContext/codebridgeContextProjectReducer.ts

```diff
@@ -76,13 +78,25 @@ export const projectReducer = (project: ProjectType, action: ReducerAction) => {
         },
       };
     }
+
+    case PROJECT_REDUCER_ACTIONS.SET_FILE_VISIBILITY: {
+      const {fileId, visibility} = <DefaultFilePayload & {visibility: boolean}>(
+        action.payload
+      );
+
+      return {
+        ...project,
+        files: {
+          ...project.files,
+          [fileId]: {...project.files[fileId], visible: visibility},
+        },
+      };
```

Comment:
I think @molly-moen and I have chatted about the fact that I've used lodash a lot more on previous teams to do deep merges like this. The syntax would be:
```
  const updatedProject = _.merge({}, project, {
    files: {
      [fileId]: {
        visible: visibility
      }
    }
  });
```
No strong feelings either way, but something to think about for deep merges!

Reply:
Right now there are around 10 functions that work in a similar way, without lodash. Since you don't have strong feelings, I think I'd prefer to handle that feedback with a follow-up task to refactor the whole file.

Reply:
Yep, no worries! And not even necessarily something to bother refactoring -- just something to keep in mind when you get into deep merging in the future.